### DISCUSSION TITLE
How do you set the 6-speed gearbox on the F355 challenge?

### DISCUSSION THREAD

**Discussion by cangarujack** [flyinghead/flycast#1295] [Q&A]
Hi,
I can't give input for 6 speed gearbox on F355.
Neither from the keyboard nor from the steering wheel.

Any help?
Thanks

**Comment by flyinghead** [comment]
Manual gearbox is mapped to Player 2 DPad:
```
    L   R
  U 1 3 5

  D 2 4 6
  ```
So 1st gear is Up+Left, 2nd gear is Down+Left, etc.

**Reply by CO-Windler** [reply]
I  found this one too. I analyzed the keyboard matrix of Sega Driving Simulator and made it mostly playable. Arcade driving games often have exotic input mappings. The emulator should support individual default mappings at different input ports (A,B,C,D). Coin slot 2 is missing. Each digital input should get an optional locking (on/off) mode (add a checkbox) to simulate rocker switches. More complex variants (like in fighting game combos) may be useful to simulate car stick shift through individual keys/buttons.
